syntax_checker.py

```diff
@@ -0,0 +1,52 @@
+import coloredlogs, logging
+import os
+import queue
+
+from pathlib import Path
+from subprocess import Popen, PIPE
+from threading import Thread
+
+# logging setup
+logger = logging.getLogger(__name__)
+coloredlogs.install(level='DEBUG', logger=logger, isatty=True, fmt='%(levelname)s %(message)s')
+
+
+class SyntaxChecker:
```

Comment:
this name could probably use some improvement

Reply:
I think it has to be pretty vague. Even vaguer than this, I think. Probably like `JobRunner` or something